### DISCUSSION TITLE
Device Share Problem

### DISCUSSION THREAD

**Discussion by ealeksandrov7** [Web3Auth/Web3Auth#813] [Help]
Hey guys.
I am using Self-Host(tkey) with react-native-sdk(v3.2.1, but tried with v3.1.0 also) for login.

![image](https://user-images.githubusercontent.com/112856139/199486155-77b4dda9-0a9e-4440-a6ff-74156a48139d.png)

When using a simulator there are no issues (doesn't matter production or dev build)
When trying the build on a physical device (iOS) I get:

![image](https://user-images.githubusercontent.com/112856139/199486957-991daf91-e538-443f-8adf-f80c98d1de7b.png)

This issue was not occurring last week.

I would really appreciate any help on this.

**Comment by shahbaz17** [comment]
Hey @ealeksandrov7 

Can you share more info about your device, browser, and login type?

**Reply by ealeksandrov7** [reply]
Device: iPhone 12
iOS Version: 15.7
Login Type: Google
Both verifier and project are 'testnet'.
For browser - I am using expo-web-browser
It is super strange that it even requires a device share.. I am trying to connect with new unused accounts.
If that helps - the error occurs on testflight builds

**Reply by ealeksandrov7** [reply]
With 'discord' login type it's the same though

**Reply by ealeksandrov7** [reply]
after further debugging with the inspector - these are the logs I get before the error pops up:

<img width="983" alt="Screen Shot 2022-11-03 at 23 29 12" src="https://user-images.githubusercontent.com/112856139/199838473-090b0c13-1ba5-46ad-8595-f03ec2fa0593.png">
<img width="993" alt="Screen Shot 2022-11-03 at 23 25 39" src="https://user-images.githubusercontent.com/112856139/199838611-a1aff6a4-964d-4e6b-87e7-806aa89f2e2e.png">

**Comment by hcm-brothers** [comment]
@ealeksandrov7 Hi there, how can you using self-host on React Native (I see web3auth doesn't support self-host on react native yet). Can you guide me how to do it?

**Reply by metallicalfa2** [reply]
Generally, the SDK will work in any JS env. Are you facing any particular build issues ?

**Reply by ratribiana** [reply]
Hi @metallicalfa2  and @hcm-brothers how is it going? Have you successfully used the self-host to your react native apps? Did you setup your own oAuth 2 server and did you add your own social media providers.?
